### PR TITLE
feat: [CDS-77535]: web-hooks sync entity pluggable drawer

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- `webhook` sync activities pluggable drawer for git entities

#### Screenshots

No Data Flow :- 

https://github.com/harness/harness-core-ui/assets/106532291/c38b4267-3418-471c-ab55-674dd452715e

Happy Flow :- 

https://github.com/harness/harness-core-ui/assets/106532291/fb3b9fb2-b7b1-4cda-817c-8d692f289cc9

Error Boundary :- 

Screenshot 2023-09-22 at 12 22 41 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/pages/webhooks/WebhookSyncDrawer/WebhookSyncDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+      event_start_time: dateFilter.startTime,
+      event_end_time: dateFilter.endTime
+    }
+  })
+  const start = new Date()
+  start.setDate(start.getDate() - 7)
+  start.setHours(0, 0, 0, 0)
+
+  const end = new Date()
+  end.setHours(23, 59, 59, 999)
```
Comment (PratyushGar) [inline src/modules/75-cd/pages/webhooks/WebhookSyncDrawer/WebhookSyncDrawer.tsx:107]:
```suggestion
const { startTime, endTime } = getDefaultDateState()
```

File: src/modules/75-cd/pages/webhooks/WebhookSyncDrawer/WebhookSyncDrawer.tsx [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    setPayload(selectedPayload)
+  }
+
+  type CustomColumn<T extends Record<string, any>> = Column<T>
+
+  const syncDrawerColumns: CustomColumn<GitXWebhookEventResponse>[] = useMemo(
+    () => [
+      {
+        Header: getString('cd.webhookEvents.dateTime').toUpperCase(),
+        id: 'datetime',
```
Comment (PratyushGar) [inline src/modules/75-cd/pages/webhooks/WebhookSyncDrawer/WebhookSyncDrawer.tsx:123]: nit: dateTime

File: src/modules/75-cd/pages/webhooks/WebhookSyncDrawer/WebhookSyncDrawer.tsx [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    try {
+      _parsedPaylod = JSON.parse(payload)
+    } catch (e) {
+      // if the parsing fails show toaster and render empty payload in the YAML builder
+      showError(e)
+      _parsedPaylod = {}
+    }
+    return _parsedPaylod
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [payload])
```
Comment (PratyushGar) [inline src/modules/75-cd/pages/webhooks/WebhookSyncDrawer/WebhookSyncDrawer.tsx:164]: showError can be added as a dependency and // eslint-disable-next-line react-hooks/exhaustive-deps can be removed

File: src/modules/75-cd/pages/webhooks/WebhookSyncDrawer/WebhookSyncDrawer.tsx [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+        Header: '',
+        id: 'payloadDetails',
+        width: '10%',
+        Cell: withWebhookEvents(WebhooksPayloadDetails),
+        actions: {
+          onClick: handlePayloadDetailsClick
+        }
+      }
+    ],
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (PratyushGar) [inline src/modules/75-cd/pages/webhooks/WebhookSyncDrawer/WebhookSyncDrawer.tsx:149]: not required

**Review by PratyushGar — DISMISSED** [review]

File: src/modules/75-cd/pages/webhooks/WebhookSyncDrawer/WebhookSyncDrawer.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/WebhookSyncDrawer/WebhookSyncDrawer.tsx]: done

File: src/modules/75-cd/pages/webhooks/WebhookSyncDrawer/WebhookSyncDrawer.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/WebhookSyncDrawer/WebhookSyncDrawer.tsx]: done

File: src/modules/75-cd/pages/webhooks/WebhookSyncDrawer/WebhookSyncDrawer.tsx (hunk 3 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/WebhookSyncDrawer/WebhookSyncDrawer.tsx]: done

File: src/modules/75-cd/pages/webhooks/WebhookSyncDrawer/WebhookSyncDrawer.tsx (hunk 4 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/WebhookSyncDrawer/WebhookSyncDrawer.tsx]: done

**Review by PratyushGar — APPROVED** [review]

**Review by Eric761 — APPROVED** [review]

**Pull request merged**